scipy/ndimage/tests/test_filters.py

```diff
@@ -1336,95 +1638,108 @@ def test_minmax_nonseparable_axes(self, filter_func, axes, kwargs):
         output = ndimage.median_filter(array, (3,))
         assert_array_almost_equal(array, output)
 
-    def test_rank03(self):
-        array = np.array([3, 2, 5, 1, 4])
+    def test_rank03(self, xp):
+        array = xp.asarray([3, 2, 5, 1, 4])
         output = ndimage.rank_filter(array, 1, size=[2])
-        assert_array_almost_equal([3, 3, 5, 5, 4], output)
+        assert_array_almost_equal(xp.asarray([3, 3, 5, 5, 4]), output)
         output = ndimage.percentile_filter(array, 100, size=2)
-        assert_array_almost_equal([3, 3, 5, 5, 4], output)
+        assert_array_almost_equal(xp.asarray([3, 3, 5, 5, 4]), output)
 
-    def test_rank04(self):
-        array = np.array([3, 2, 5, 1, 4])
-        expected = [3, 3, 2, 4, 4]
+    def test_rank04(self, xp):
+        array = xp.asarray([3, 2, 5, 1, 4])
+        expected = xp.asarray([3, 3, 2, 4, 4])
         output = ndimage.rank_filter(array, 1, size=3)
         assert_array_almost_equal(expected, output)
         output = ndimage.percentile_filter(array, 50, size=3)
         assert_array_almost_equal(expected, output)
         output = ndimage.median_filter(array, size=3)
         assert_array_almost_equal(expected, output)
 
-    def test_rank05(self):
-        array = np.array([3, 2, 5, 1, 4])
-        expected = [3, 3, 2, 4, 4]
+    def test_rank05(self, xp):
+        array = xp.asarray([3, 2, 5, 1, 4])
+        expected = xp.asarray([3, 3, 2, 4, 4])
         output = ndimage.rank_filter(array, -2, size=3)
         assert_array_almost_equal(expected, output)
 
-    def test_rank06(self):
-        array = np.array([[3, 2, 5, 1, 4],
-                          [5, 8, 3, 7, 1],
-                          [5, 6, 9, 3, 5]])
+    def test_rank06(self, xp):
+        array = xp.asarray([[3, 2, 5, 1, 4],
+                            [5, 8, 3, 7, 1],
+                            [5, 6, 9, 3, 5]])
         expected = [[2, 2, 1, 1, 1],
                     [3, 3, 2, 1, 1],
                     [5, 5, 3, 3, 1]]
+        expected = xp.asarray(expected)
         output = ndimage.rank_filter(array, 1, size=[2, 3])
         assert_array_almost_equal(expected, output)
         output = ndimage.percentile_filter(array, 17, size=(2, 3))
         assert_array_almost_equal(expected, output)
 
-    def test_rank06_overlap(self):
-        array = np.array([[3, 2, 5, 1, 4],
-                          [5, 8, 3, 7, 1],
-                          [5, 6, 9, 3, 5]])
-        array_copy = array.copy()
+    @skip_xp_backends("jax.numpy",
+        reasons=["assignment destination is read-only"]
+    )
+    def test_rank06_overlap(self, xp):
+        if is_cupy(xp):
+            pytest.xfail("https://github.com/cupy/cupy/issues/8406")
+        array = xp.asarray([[3, 2, 5, 1, 4],
+                            [5, 8, 3, 7, 1],
+                            [5, 6, 9, 3, 5]])
+
+        asarray = array_namespace(array).asarray
+        array_copy = asarray(array, copy=True)
```

Comment:
we can use `_lib._array_api._asarray` here instead

Reply:
Is the present version bad somehow. It works across the CI + locally on GPU, it looks obviously correct from a quick glance, what is wrong with it? 
A suggested alternative has three leading underscores, and I've no idea without testing what it's going to do under plain numpy.

Reply:
I don't think the present version is "bad", it's just "not quite as good" as using the helpers which are used elsewhere to abstract away the `array_namespace` call. Recalling the argument along the lines of 'some maintainers don't care about this stuff, so let's try to be minimally intrusive in the code'. This isn't a particularly good example since your pattern was short anyway, but I think it's generally good practice to use existing abstractions rather than reinvent the wheel.

> A suggested alternative has three leading underscores, and I've no idea without testing what it's going to do under plain numpy.

Well `_lib._array_api.array_namespace` has 2 leading underscores, so I don't really see the issue 🙃. If you would like to know what the function will do, take a look at [the docstring](https://github.com/scipy/scipy/blob/a58f4b61557b9ca649fd63d05338eee368efce41/scipy/_lib/_array_api.py#L132-L155)!

Reply:
All that to say that I don't think you need to change anything for me to merge this - just trying to encourage best practice.